cincinnati/src/plugins/internal/graph_builder/openshift_secondary_metadata_parser/plugin.rs

```diff
@@ -478,6 +491,52 @@ impl OpenshiftSecondaryMetadataParserPlugin {
         Ok(())
     }
 
+    async fn process_conditional_edges(
+        &self,
+        graph: &mut cincinnati::Graph,
+        data_dir: &PathBuf,
+    ) -> Fallible<()> {
+        let blocked_edges_dir = data_dir.join(BLOCKED_EDGES_DIR);
+        let conditional_edges: Vec<graph_data_model::ConditionalEdgeYaml> =
+            deserialize_directory_files(
+                &blocked_edges_dir,
+                regex::Regex::new("ya+ml")?,
+                &self.settings.disallowed_errors,
+            )
+            .await
+            .context(format!(
+                "looking for conditional edges in blocked edges {:?}",
+                blocked_edges_dir
+            ))?;
+
+        debug!(
+            "Found {} valid conditional edges declarations.",
+            conditional_edges.len()
+        );
+
+        conditional_edges
+            .into_iter()
+            .try_for_each(|cey| -> Fallible<()> {
+                let ce: ConditionalEdge = ConditionalEdge {
+                    edge_regex: ConditionalUpdateEdge {
+                        from: cey.from.to_string(),
+                        to: cey.to.to_string(),
+                    },
+                    edges: vec![],
+                    risks: vec![ConditionalUpdateRisk {
```

Comment:
We should add a TODO here that some edges could be blocked on several risks

Reply:
makes sense, I've left the work to remove edge-risk duplication for later. Added a TODO here.